src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheckTest.java

```diff
@@ -62,10 +63,7 @@ public void testWithDefault()
             throws Exception {
         final DefaultConfiguration checkConfig =
             createModuleConfig(IllegalImportCheck.class);
-        final String[] expected = {
-            "15:1: " + getCheckMessage(MSG_KEY, "sun.applet.*"),
-            "28:1: " + getCheckMessage(MSG_KEY, "sun.*"),
-        };
+        final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;
```

Comment:
Violations shouldn't really be changing.
Also since you remove all checks of `sun` for `IllegalImportCheck`, I think pitest will complain about no coverage for https://github.com/checkstyle/checkstyle/blob/90f20e09869c71eb22190ad9c1d46d5deec324a5/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheck.java#L98 .

Most likely this will have to be reverted to go to the non-compilable folder.

Reply:
Pitest did fail.

Reply:
Maybe we will add our own packages for import tests? I mean

```
src
  |-test
      |-resources
          |- com.puppycrowl.tools
          |- com.special.package.for.import.tests
          |- org.special.package.for.import.tests
```
This will satisfy #4845. But in the test tree there will be mess.

Reply:
> But in the test tree there will be mess.

it is better to not do this.
Just copy problematic imports to non-compilable folder. ImportOrderCheck already have a lot of Input files in non-compilable folder  because we need a lot of cases with names and we can not afford to keep all that in out compilation dependencies.